### ISSUE TITLE
Problem with jumping molecules (again)

### ISSUE THREAD

**Issue by leeping** [openmm/openmm#4125]
Unfortunately, I think I saw this error rear its head again in commit 5e9134005d3ca013572979289ce00ec32038e4f1.  It only occurred for the simulation with a very large number for the barostat attempt interval (once every 500 steps).  Notably, it does not occur when the barostat attempt interval is set to every 50 steps with everything else being the same.

After running for about 20 ns, the simulation with the jumps will segfault.  It also produces a core dump but gdb does not print anything useful (presumably because OpenMM wasn't compiled with debug flags).

The error message from the segfault was:
```
Traceback (most recent call last):
  File "/home/leeping/src/OpenMM-MD/OpenMM-MD.py", line 1611, in <module>
    simulation.step(args.production)
  File "/home/leeping/opt/miniconda3/envs/openmm_80_fixmc/lib/python3.9/site-packages/openmm/app/simulation.py", line 141, in step
    self._simulate(endStep=self.currentStep+steps)
  File "/home/leeping/opt/miniconda3/envs/openmm_80_fixmc/lib/python3.9/site-packages/openmm/app/simulation.py", line 206, in _simulate
    self.integrator.step(10) # Only take 10 steps at a time, to give Python more chances to respond to a control-c.
  File "/home/leeping/opt/miniconda3/envs/openmm_80_fixmc/lib/python3.9/site-packages/openmm/openmm.py", line 9263, in step
    return _openmm.CustomIntegrator_step(self, steps)
openmm.OpenMMException: Error invoking kernel: CUDA_ERROR_ILLEGAL_ADDRESS (700)
```

I've attached the scripts to reproduce.  The first jump occurs within the first 0.5 ns or so.

[jump_molecule2.tar.gz](https://github.com/openmm/openmm/files/11864248/jump_molecule2.tar.gz)

_Originally posted by @leeping in https://github.com/openmm/openmm/issues/4117#issuecomment-1606710176_

**Comment by sef43** [comment]
I am running your script. It has got to 60ns without segfaulting, will keep it going and report back. I also don't think I am seeing the atom indices being scrambled. How are you detecting the jumps?

**Comment by leeping** [comment]
Thanks for looking into this.  I ran nine simulations in a script, and only the two with the large barostat attempt interval segfaulted.  I'll try to reproduce it to ensure I didn't make a mistake.

I'm detecting the jumps by visualizing the trajectory in VMD and selecting the O atoms in one half of the box at a given frame (i.e. `name O and x < 12` or something).  I then animate the trajectory without updating the selection.  When the behavior is normal the molecules diffuse very slowly out of the "half box", but if there is a jump at a given trajectory frame, they appear all over the box after advancing one frame.  Admittedly it's a bit lazy, and I should really be writing a script to plot the MSD as a function of time.

**Comment by sef43** [comment]
My run of your script crashed after ~80ns. The specific reason for the crash seems to be the particle positions becoming very large, so consistent with the typical force field instability style crash. Specific error message was:
```
Traceback (most recent call last):
  File "/home/sfarr/Documents/openmm_issues/issue_4125/jump_molecule2/OpenMM-MD.py", line 1611, in <module>
    simulation.step(args.production)
  File "/home/sfarr/miniconda3/envs/tempnew/lib/python3.11/site-packages/openmm/app/simulation.py", line 141, in step
    self._simulate(endStep=self.currentStep+steps)
  File "/home/sfarr/miniconda3/envs/tempnew/lib/python3.11/site-packages/openmm/app/simulation.py", line 243, in _simulate
    self._generate_reports(unwrapped, False)
  File "/home/sfarr/miniconda3/envs/tempnew/lib/python3.11/site-packages/openmm/app/simulation.py", line 263, in _generate_reports
    reporter.report(self, state)
  File "/home/sfarr/miniconda3/envs/tempnew/lib/python3.11/site-packages/openmm/app/pdbreporter.py", line 118, in report
    PDBFile.writeModel(topology, positions, self._out, self._nextModel)
  File "/home/sfarr/miniconda3/envs/tempnew/lib/python3.11/site-packages/openmm/app/pdbfile.py", line 388, in writeModel
    recordName, _formatIndex(atomIndex, 5), atomName, resName, chainName, resId, resIC, _format_83(coords[0]),
                                                                                        ^^^^^^^^^^^^^^^^^^^^^
  File "/home/sfarr/miniconda3/envs/tempnew/lib/python3.11/site-packages/openmm/app/pdbfile.py", line 471, in _format_83
    raise ValueError('coordinate "%s" could not be represented '
ValueError: coordinate "167514310.32176882" could not be represented in a width-8 field
```
Note I changed dcd reporter to pdb reporter for using different visualization software.

Looking at the reported temperature it seems the thermostat is struggling to effectively maintain constant temperature:
<img width="764" alt="Screenshot 2023-06-27 at 17 14 05" src="https://github.com/openmm/openmm/assets/25941757/9d09420a-d367-4f9a-85d4-db8447688418">

Looking at the trajectory I am not seeing atom indices being incorrectly reordered as seen in the previous issue but rather larger atom displacements consistent with large unstable velocities. 

I am guessing the  `CUDA_ERROR_ILLEGAL_ADDRESS (700)` error you are seeing could be caused by large unstable velocities/positions.

I think this could be a different issue perhaps caused by the simulation parameters. Would you expect a barostat with such a high interval between attempts to be able to effectively control pressure? The default is 25 timesteps. I would be interested to see what happens with a simple 0.5fs langevinMiddleIntegrator instead of the custom multiple timestep one.

**Comment by peastman** [comment]
A very large interval isn't exactly "wrong".  Given enough time, it will still sample the correct distribution.  But it will sample it very slowly.  And if something happens that should cause the box size to change (like a folding or unfolding event), it will take a long time to respond.

**Comment by leeping** [comment]
I observed the issue again, this time after about 1 μs of simulation time (about 1 week). The barostat interval was 50 steps and the thermostat collision frequency was 0.01 ps^-1.  (I also observed the same thing with a 0.1 ps^-1 collision frequency, but I did not have a restart file to reproduce the error).

The files to reproduce the issue, along with three saved runs, are located under `reproduce_crash_2/template` in the attached archive.  The restart file comes from a run at 188.15 K, 1600.0 atm with a 0.01 ps^-1 collision frequency, 50-step barostat interval, and 1.0 fs time step, at about 900 ns.  There is about a 5% chance of observing the simulation blow up when restarting from the restart file.  

The temperature is already elevated at the start of the "restart", which indicates that some artifact has occurred that the thermostat is failing to damp away.  I also attach a movie that shows what happens immediately before the blowing up; it appears to be some kind of rotation at around frame 18010 of the trajectory in `reproduce_crash_2/saved_2`.  Note that I never observed this when using OpenMM 7.4 or 7.5; I am able to simulate for multiple μs using those versions with no crashes.

The movie: https://github.com/openmm/openmm/assets/1441560/b0b9d6ac-eabe-46f0-92c0-b4c82d128fc3

The archive: https://www.dropbox.com/scl/fi/6t6bhccj9i07pnmp5ef2p/reproduce_crash_2.tar.gz?rlkey=nh5zaxy8vkzk7wd4vqu2b7ri4&dl=0

**Comment by leeping** [comment]
I found a quicker way to reproduce the issue. You can start from the input file in `reproduce_crash_2/template` and change the collision frequency to 1e-10. The result will be a simulation that approximately conserves energy between barostat steps. When the barostat does take a step, the change in the total energy should be small.

The attached screenshot plots the total energy vs. time.  You can see that in the result using https://github.com/openmm/openmm/commit/5e9134005d3ca013572979289ce00ec32038e4f1, the energy increases and then suddenly blows up at 2.183 ps.  But the last barostat volume adjustment was at 2.100 ps, so the times don't coincide.  

I have not observed the simulation blowing up using OpenMM 7.4, or in a NVT run using either version, but I'll let you know if that changes.

![image](https://github.com/openmm/openmm/assets/1441560/ff4ff73c-dda7-4a1d-8509-f262d183cb23)

![image](https://github.com/openmm/openmm/assets/1441560/8b9f7835-bf41-4f9a-ab90-eea7c544b5e3)

**Comment by peastman** [comment]
If the blowup doesn't coincide with the barostat step, that suggests this isn't related to the barostat?

**Comment by leeping** [comment]
At the same time, I’ve yet to encounter the issue in any simulation where
the barostat is turned off (but I’ll let you know if I do.)

**Comment by leeping** [comment]
Following up on before: I have some evidence to suggest the simulation blow-up is still related to the barostat.  From a previous post I mentioned I had observed some strange accelerations in the system prior to the blow-up.  I calculated the accelerations from the trajectory using central finite difference of the positions, then multiplied by the atomic masses to convert to forces.  I then used OpenMM to compute the force on the central frame and compared them to the forces estimated from acceleration.  The below output shows the trajectory frame number, the maximum absolute difference between the two sets of forces, and the array index of the maximum absolute force difference.

Before step 18000, the maximum force error between the two sets of forces is on the order of 1e2 kJ/mol/nm and the index with the maximum value seems random.  However, after step 18000 (where a successful MC occurs changing the density from 1037 -> 1062 kg m^-3), the maximum force error increases to more than 1e3 kJ/mol/nm and it is persistently on a single component (2423).  If the MC move is generating an erroneous force contribution, it may not cause the simulation to blow up immediately, but instead could give rise to integrator errors that lead to the blow-up a short time later.  This erroneous force contribution reappears several times over the whole trajectory but it does not cause a blow-up until around frame 18050.

I can provide more analysis (such as visualizing the direction of the force differences) if that would help.

```
 #Step  MaxFrcErr Argmax
 17971 1.4660e+02   1198
 17972 9.9374e+01   3159
 17973 1.2987e+02   2846
 17974 1.1814e+02    640
 17975 1.1358e+02   2882
 17976 9.9064e+01    812
 17977 1.2320e+02   2376
 17978 9.8162e+01    865
 17979 9.8455e+01   1008
 17980 1.1446e+02    244
 17981 1.0098e+02   3943
 17982 1.0395e+02   3701
 17983 9.7130e+01   3503
 17984 1.0277e+02   1991
 17985 9.7147e+01   4151
 17986 9.9938e+01   1917
 17987 1.0484e+02   3240
 17988 9.9023e+01   2692
 17989 9.8576e+01   2675
 17990 1.0956e+02    279
 17991 1.0264e+02    686
 17992 1.1443e+02   1721
 17993 1.0968e+02   3863
 17994 1.0814e+02   4374
 17995 1.1001e+02    810
 17996 9.9029e+01    875
Skipping frame 17999 due to MC barostat move
Skipping frame 18000 due to MC barostat move
Skipping frame 18001 due to MC barostat move
Skipping frame 18002 due to MC barostat move
 18001 2.7491e+03   2423
 18002 2.5889e+03   2423
 18003 2.5969e+03   2423
 18004 2.5375e+03   2423
 18005 2.6194e+03   2423
 18006 2.5367e+03   2423
 18007 2.4658e+03   2423
 18008 2.4051e+03   2423
 18009 2.3582e+03   2423
 18010 2.1906e+03   2423
 18011 2.2351e+03   2423
 18012 2.5640e+03   2423
 18013 3.3264e+03   2423
 18014 4.3840e+03   2423
 18015 5.2587e+03   2423
 18016 6.4825e+03   2423
 18017 7.4362e+03   2423
 18018 8.0897e+03   2423
 18019 8.1141e+03   2423
 18020 7.7886e+03   2423
 18021 7.4643e+03   2423
 18022 8.5428e+03   2421
 18023 9.9918e+03   2421
 18024 1.0880e+04   2421
 18025 1.0483e+04   2421
 18026 8.6859e+03   2421
 18027 6.0595e+03   2421
 18028 3.2624e+03   2423
 18029 2.3077e+03   2423
 18030 3.3205e+03   2424
 18031 3.5399e+03   2426
 18032 2.0786e+03   2426
 18033 2.1432e+03   2426
 18034 3.0079e+03   2424
 18035 1.2990e+03   2424
 18036 1.3460e+03   2421
 18037 1.1649e+03   2421
 18038 9.0517e+02   2421
 18039 6.5281e+02   2423
 18040 5.1951e+02   2423
 18041 3.3113e+02   2423
 18042 5.1624e+02   1383
 18043 1.2572e+03   1384
 18044 4.0232e+03   1383
 18045 3.8585e+04   2421
 18046 1.4068e+05   1383
```

**Comment by peastman** [comment]
Do you have a way to reproduce this?  We need to figure out where the acceleration is coming from.  It seems the forces computed during the simulation are different from the ones you compute afterward from the coordinates?  Why?

The behavior in your video is strange.  It seems to affect just one water molecule, while all the ones around it continue to behave normally.  If this were another issue with atom reordering, I would expect it to affect many molecules, not just one.  And it clearly isn't something that happens only during the step in which the barostat acts, because the incorrect acceleration continues for many steps afterward.  And it grows with time.

I wonder if it could be a memory error, for example if the charge on an atom is getting overwritten with the wrong value?  But then we would expect to see errors on nearby atoms, which we don't.

**Comment by sef43** [comment]
I can reproduce this with the following script (simplified version of @leeping's script with similar settings), the files needed to run it are attached in the Archive.zip folder (waterbox.pdb, iamoeba.xml, restart.p)  :
```python
from openmm.app import *
from openmm import *
from openmm.unit import *
from sys import stdout
import pickle

pdb = PDBFile('waterbox.pdb')
forcefield = ForceField('iamoeba.xml')

settings_dict={
'ewaldErrorTolerance'      : 3e-05, 
'nonbondedCutoff'          : 0.7*nanometer, 
'nonbondedMethod'          : PME, 
'polarization'             : 'direct', 
'removeCMMotion'           : False,
'rigidWater'               : False, 
'useDis
[Archive.zip](https://github.com/openmm/openmm/files/12293059/Archive.zip)
persionCorrection'  : True, 
'vdwCutoff'                : 0.9*nanometer} 

timestep = 0.5*femtosecond
collision_frequency = 0.01/picosecond
barostat_frquency = 10
temperature = 188.15*kelvin
pressure = 1600*atmosphere

system = forcefield.createSystem(pdb.topology, **settings_dict)

integrator = LangevinMiddleIntegrator(temperature, collision_frequency, timestep)

simulation = Simulation(pdb.topology, system, integrator)

r_positions, r_velocities, r_boxes = pickle.load(open('restart.p', 'rb'))

simulation.context.setPeriodicBoxVectors(r_boxes[0] * nanometer,r_boxes[1] * nanometer, r_boxes[2] * nanometer)
simulation.context.setPositions(r_positions * nanometer)
simulation.context.setVelocities(r_velocities * nanometer / picosecond)

system.addForce(MonteCarloBarostat(pressure, temperature, barostat_frquency))
simulation.context.reinitialize(preserveState=True)


for i,force in enumerate(system.getForces()):
    force.setForceGroup(i)
    i+=1


print("Minimizing energy...")
simulation.minimizeEnergy(maxIterations=100)

simulation.reporters.append(DCDReporter('output.dcd', 1000))
simulation.reporters.append(StateDataReporter(stdout, 1000, step=True,
        potentialEnergy=True, temperature=True, speed=True, volume=True))


print("Running simulation...")

for t in range(1000):

    simulation.step(1000)

    for i, f, in enumerate(system.getForces()):
        print(f.getName(), simulation.context.getState(getEnergy=True, groups={i}).getPotentialEnergy())
        
```
It does not always crash but if I run it ~5 times one of them will crash with NAN after approximately 20,000 steps. E.g. here is the output:

<details>

```
Minimizing energy...
Running simulation...
#"Step","Potential Energy (kJ/mole)","Temperature (K)","Box Volume (nm^3)","Speed (ns/day)"
1000,-24475.913109486446,118.2498260868666,13.533177350173744,0
AmoebaBond 2308.7337036132812 kJ/mol
AmoebaVdwForce 8382.357636267798 kJ/mol
AmoebaMultipoleForce -35167.00383758545 kJ/mol
HarmonicBondForce 0.0 kJ/mol
AmoebaAngle 0.0 kJ/mol
AmoebaInPlaneAngle 0.0 kJ/mol
MonteCarloBarostat 0.0 kJ/mol
2000,-24441.20407315318,118.68115184659086,13.671349111446453,80.7
AmoebaBond 2249.3948974609375 kJ/mol
AmoebaVdwForce 8127.226868428546 kJ/mol
AmoebaMultipoleForce -34817.82543182373 kJ/mol
HarmonicBondForce 0.0 kJ/mol
AmoebaAngle 0.0 kJ/mol
AmoebaInPlaneAngle 0.0 kJ/mol
MonteCarloBarostat 0.0 kJ/mol
3000,-24463.202906006914,120.6791887625677,13.57270205865948,82.4
AmoebaBond 2455.9127197265625 kJ/mol
AmoebaVdwForce 8370.419179087537 kJ/mol
AmoebaMultipoleForce -35289.53474998474 kJ/mol
HarmonicBondForce 0.0 kJ/mol
AmoebaAngle 0.0 kJ/mol
AmoebaInPlaneAngle 0.0 kJ/mol
MonteCarloBarostat 0.0 kJ/mol
4000,-24469.878797946407,121.8719962102913,13.659661750389386,82.7
AmoebaBond 2304.93017578125 kJ/mol
AmoebaVdwForce 8178.881755413579 kJ/mol
AmoebaMultipoleForce -34953.6902923584 kJ/mol
HarmonicBondForce 0.0 kJ/mol
AmoebaAngle 0.0 kJ/mol
AmoebaInPlaneAngle 0.0 kJ/mol
MonteCarloBarostat 0.0 kJ/mol
5000,-24465.26614291294,121.84843678648258,13.584869850637283,83
AmoebaBond 2267.490966796875 kJ/mol
AmoebaVdwForce 8451.771293572348 kJ/mol
AmoebaMultipoleForce -35184.528440475464 kJ/mol
HarmonicBondForce 0.0 kJ/mol
AmoebaAngle 0.0 kJ/mol
AmoebaInPlaneAngle 0.0 kJ/mol
MonteCarloBarostat 0.0 kJ/mol
6000,-24397.031632386625,119.63898902789529,13.488648055433831,83.5
AmoebaBond 2437.9534912109375 kJ/mol
AmoebaVdwForce 8870.804763830723 kJ/mol
AmoebaMultipoleForce -35705.789887070656 kJ/mol
HarmonicBondForce 0.0 kJ/mol
AmoebaAngle 0.0 kJ/mol
AmoebaInPlaneAngle 0.0 kJ/mol
MonteCarloBarostat 0.0 kJ/mol
7000,-24472.83461263578,124.3918130645429,13.49779829769628,82.8
AmoebaBond 2410.2520751953125 kJ/mol
AmoebaVdwForce 8739.703771801782 kJ/mol
AmoebaMultipoleForce -35622.790897369385 kJ/mol
HarmonicBondForce 0.0 kJ/mol
AmoebaAngle 0.0 kJ/mol
AmoebaInPlaneAngle 0.0 kJ/mol
MonteCarloBarostat 0.0 kJ/mol
8000,-24421.50368417991,122.11568129963034,13.635483720429946,82.9
AmoebaBond 2393.3458251953125 kJ/mol
AmoebaVdwForce 8465.949180228587 kJ/mol
AmoebaMultipoleForce -35280.79875564575 kJ/mol
HarmonicBondForce 0.0 kJ/mol
AmoebaAngle 0.0 kJ/mol
AmoebaInPlaneAngle 0.0 kJ/mol
MonteCarloBarostat 0.0 kJ/mol
9000,-24443.877683124254,125.02524715973892,13.634063561138719,83.1
AmoebaBond 2387.0210571289062 kJ/mol
AmoebaVdwForce 8485.822898426342 kJ/mol
AmoebaMultipoleForce -35316.72137117386 kJ/mol
HarmonicBondForce 0.0 kJ/mol
AmoebaAngle 0.0 kJ/mol
AmoebaInPlaneAngle 0.0 kJ/mol
MonteCarloBarostat 0.0 kJ/mol
10000,-24413.396890148604,122.7277212226693,13.601771636510747,83.4
AmoebaBond 2323.8088989257812 kJ/mol
AmoebaVdwForce 8491.715564265764 kJ/mol
AmoebaMultipoleForce -35228.92116737366 kJ/mol
HarmonicBondForce 0.0 kJ/mol
AmoebaAngle 0.0 kJ/mol
AmoebaInPlaneAngle 0.0 kJ/mol
MonteCarloBarostat 0.0 kJ/mol
11000,-24456.095598538228,124.41954773729768,13.541518459812437,83.6
AmoebaBond 2405.8884887695312 kJ/mol
AmoebaVdwForce 8629.650408904245 kJ/mol
AmoebaMultipoleForce -35491.63485622406 kJ/mol
HarmonicBondForce 0.0 kJ/mol
AmoebaAngle 0.0 kJ/mol
AmoebaInPlaneAngle 0.0 kJ/mol
MonteCarloBarostat 0.0 kJ/mol
12000,-24481.887824804217,127.32163370318119,13.533430849568012,84
AmoebaBond 2509.341552734375 kJ/mol
AmoebaVdwForce 8712.819679945083 kJ/mol
AmoebaMultipoleForce -35704.04895019531 kJ/mol
HarmonicBondForce 0.0 kJ/mol
AmoebaAngle 0.0 kJ/mol
AmoebaInPlaneAngle 0.0 kJ/mol
MonteCarloBarostat 0.0 kJ/mol
13000,-24439.181699712317,124.66360401456218,13.590509560338377,84.2
AmoebaBond 2344.4425048828125 kJ/mol
AmoebaVdwForce 8607.918558638055 kJ/mol
AmoebaMultipoleForce -35391.54254102707 kJ/mol
HarmonicBondForce 0.0 kJ/mol
AmoebaAngle 0.0 kJ/mol
AmoebaInPlaneAngle 0.0 kJ/mol
MonteCarloBarostat 0.0 kJ/mol
14000,-24428.563034910854,125.24298423941771,13.596358099365723,84.4
AmoebaBond 2375.38427734375 kJ/mol
AmoebaVdwForce 8452.045116524996 kJ/mol
AmoebaMultipoleForce -35255.99171447754 kJ/mol
HarmonicBondForce 0.0 kJ/mol
AmoebaAngle 0.0 kJ/mol
AmoebaInPlaneAngle 0.0 kJ/mol
MonteCarloBarostat 0.0 kJ/mol
15000,-24410.08575702043,125.90210611765853,13.534097829576687,84.5
AmoebaBond 2427.6732177734375 kJ/mol
AmoebaVdwForce 8573.214471098467 kJ/mol
AmoebaMultipoleForce -35410.974093437195 kJ/mol
HarmonicBondForce 0.0 kJ/mol
AmoebaAngle 0.0 kJ/mol
AmoebaInPlaneAngle 0.0 kJ/mol
MonteCarloBarostat 0.0 kJ/mol
16000,-24471.326515238616,129.16543524194282,13.524632786025172,84.5
AmoebaBond 2447.1852416992188 kJ/mol
AmoebaVdwForce 8721.551146466403 kJ/mol
AmoebaMultipoleForce -35640.06250190735 kJ/mol
HarmonicBondForce 0.0 kJ/mol
AmoebaAngle 0.0 kJ/mol
AmoebaInPlaneAngle 0.0 kJ/mol
MonteCarloBarostat 0.0 kJ/mol
17000,-24443.843935988032,129.02377328835703,13.530643221579092,84.3
AmoebaBond 2531.02099609375 kJ/mol
AmoebaVdwForce 8889.594481446658 kJ/mol
AmoebaMultipoleForce -35864.459215164185 kJ/mol
HarmonicBondForce 0.0 kJ/mol
AmoebaAngle 0.0 kJ/mol
AmoebaInPlaneAngle 0.0 kJ/mol
MonteCarloBarostat 0.0 kJ/mol
18000,-24446.268534709965,129.18229291444217,13.493115422880088,84.2
AmoebaBond 2437.54248046875 kJ/mol
AmoebaVdwForce 8940.529397914852 kJ/mol
AmoebaMultipoleForce -35824.34015274048 kJ/mol
HarmonicBondForce 0.0 kJ/mol
AmoebaAngle 0.0 kJ/mol
AmoebaInPlaneAngle 0.0 kJ/mol
MonteCarloBarostat 0.0 kJ/mol
19000,-24409.348997772133,128.2661009217774,13.691695786992696,84.2
AmoebaBond 2358.692138671875 kJ/mol
AmoebaVdwForce 8312.352813541494 kJ/mol
AmoebaMultipoleForce -35080.394229888916 kJ/mol
HarmonicBondForce 0.0 kJ/mol
AmoebaAngle 0.0 kJ/mol
AmoebaInPlaneAngle 0.0 kJ/mol
MonteCarloBarostat 0.0 kJ/mol
20000,-24379.601731189257,126.81507070590479,13.649391976674263,84.2
AmoebaBond 2333.6202392578125 kJ/mol
AmoebaVdwForce 8371.59785377168 kJ/mol
AmoebaMultipoleForce -35084.82015609741 kJ/mol
HarmonicBondForce 0.0 kJ/mol
AmoebaAngle 0.0 kJ/mol
AmoebaInPlaneAngle 0.0 kJ/mol
MonteCarloBarostat 0.0 kJ/mol
21000,-24449.088155343336,131.24484389062974,13.651857942328336,84.1
AmoebaBond 2398.0139770507812 kJ/mol
AmoebaVdwForce 8319.407174990376 kJ/mol
AmoebaMultipoleForce -35166.50877380371 kJ/mol
HarmonicBondForce 0.0 kJ/mol
AmoebaAngle 0.0 kJ/mol
AmoebaInPlaneAngle 0.0 kJ/mol
MonteCarloBarostat 0.0 kJ/mol
22000,-24396.65155414796,129.29455570983387,13.694794774362471,83.9
AmoebaBond 2344.61669921875 kJ/mol
AmoebaVdwForce 8253.239391286608 kJ/mol
AmoebaMultipoleForce -34994.50753211975 kJ/mol
HarmonicBondForce 0.0 kJ/mol
AmoebaAngle 0.0 kJ/mol
AmoebaInPlaneAngle 0.0 kJ/mol
MonteCarloBarostat 0.0 kJ/mol
23000,-24404.903245330002,129.90760152402203,13.643059816830156,83.8
AmoebaBond 2491.5730590820312 kJ/mol
AmoebaVdwForce 8381.407245397424 kJ/mol
AmoebaMultipoleForce -35277.88344955444 kJ/mol
HarmonicBondForce 0.0 kJ/mol
AmoebaAngle 0.0 kJ/mol
AmoebaInPlaneAngle 0.0 kJ/mol
MonteCarloBarostat 0.0 kJ/mol
24000,-24352.998784106738,127.83942799072781,13.699519090445326,83.8
AmoebaBond 2617.0928955078125 kJ/mol
AmoebaVdwForce 8380.127415138715 kJ/mol
AmoebaMultipoleForce -35350.21890211105 kJ/mol
HarmonicBondForce 0.0 kJ/mol
AmoebaAngle 0.0 kJ/mol
AmoebaInPlaneAngle 0.0 kJ/mol
MonteCarloBarostat 0.0 kJ/mol
25000,-23612.29073411499,159.61080964894293,13.737158018905403,83.8
AmoebaBond 2381.4647216796875 kJ/mol
AmoebaVdwForce 8228.41908567871 kJ/mol
AmoebaMultipoleForce -34222.17466163635 kJ/mol
HarmonicBondForce 0.0 kJ/mol
AmoebaAngle 0.0 kJ/mol
AmoebaInPlaneAngle 0.0 kJ/mol
MonteCarloBarostat 0.0 kJ/mol
26000,4.1192185555262484e+33,1694559347245997.8,13.204841637386833,82.3
AmoebaBond 4.1192185555262484e+33 kJ/mol
AmoebaVdwForce 9869501.380326143 kJ/mol
AmoebaMultipoleForce -69346937.95537567 kJ/mol
HarmonicBondForce 0.0 kJ/mol
AmoebaAngle 0.0 kJ/mol
AmoebaInPlaneAngle 0.0 kJ/mol
MonteCarloBarostat 0.0 kJ/mol
27000,7.972763467709083e+35,2667068291783621.5,11.178586042669005,80.4
AmoebaBond 7.972763467709083e+35 kJ/mol
AmoebaVdwForce 51262489.81172352 kJ/mol
AmoebaMultipoleForce -296069249.3634033 kJ/mol
HarmonicBondForce 0.0 kJ/mol
AmoebaAngle 0.0 kJ/mol
AmoebaInPlaneAngle 0.0 kJ/mol
MonteCarloBarostat 0.0 kJ/mol
28000,1.104952177539832e+37,2572808603574318.5,10.078218448041092,78.7
AmoebaBond 1.104952177539832e+37 kJ/mol
AmoebaVdwForce 6128940.25502598 kJ/mol
AmoebaMultipoleForce -948617.474319458 kJ/mol
HarmonicBondForce 0.0 kJ/mol
AmoebaAngle 0.0 kJ/mol
AmoebaInPlaneAngle 0.0 kJ/mol
MonteCarloBarostat 0.0 kJ/mol
29000,7.070840959519043e+37,2497502213116018.0,8.954719485317,77.6
AmoebaBond 7.070840959519043e+37 kJ/mol
AmoebaVdwForce 37955728.47451019 kJ/mol
AmoebaMultipoleForce -1192791947.4796677 kJ/mol
HarmonicBondForce 0.0 kJ/mol
AmoebaAngle 0.0 kJ/mol
AmoebaInPlaneAngle 0.0 kJ/mol
MonteCarloBarostat 0.0 kJ/mol
30000,3.0394936092418422e+38,2183238280196321.2,8.358279111895161,76.6
AmoebaBond 3.0394936092418422e+38 kJ/mol
AmoebaVdwForce 18287952.41068804 kJ/mol
AmoebaMultipoleForce 437674.4701538086 kJ/mol
HarmonicBondForce 0.0 kJ/mol
AmoebaAngle 0.0 kJ/mol
AmoebaInPlaneAngle 0.0 kJ/mol
MonteCarloBarostat 0.0 kJ/mol


```

</details>

**Comment by leeping** [comment]
Thanks for both of your efforts to continue to look into this.  I think the issue may not be due to incorrect potential parameters, because if I take the forces calculated from the trajectory and sum over all atoms, the result is pretty large.  That indicates maybe the calculations of accelerations (if there's any force-independent part) or the position updates are not being done correctly.  Or could it be an error in the calculation of torque, or conversion of torque to force?

**Comment by sef43** [comment]
I also notice that If I increase the nonbonded cutoff to 1.0 nanometers then the crash happens much more quickly and on every repeat.

**Comment by peastman** [comment]
This is definitely looking like memory corruption of some sort.  I've narrowed it down to a particular kernel launch, after which the torques on certain particles have huge values.  But that kernel doesn't even have a reference to the array containing torques...

**Comment by sef43** [comment]
@peastman out of curiosity do you use a specific debugging program to query the values in the force/toque arrays? or do you just use good old print statements?

**Comment by peastman** [comment]
Just `printf()`.

Here's another method that's useful for cases like this.  The script above is slow and nondeterministic.  It usually takes thousands of steps to reproduce the problem, and you never know how many steps it will take or which particles it will appear in.  That's hard to debug.  To make a better test case, create a checkpoint after every step.  Keep around a few of the most recent ones.  When the problem appears, as signaled by a sudden jump in the energy, save a checkpoint from a few steps back.

```python
for j in range(steps):
    checkpoints.append(simulation.context.createCheckpoint())
    if len(checkpoints) > 5:
        del checkpoints[0]
    simulation.step(1)
    e = simulation.context.getState(getEnergy=True, groups={0}).getPotentialEnergy()._value
    if e > 1e4:
        with open('checkpoint', 'wb') as f:
            f.write(checkpoints[0])
        exit()
```

Now you can load in the checkpoint and deterministically reproduce the problem and the last few steps leading up to it.

**Comment by peastman** [comment]
The fix is in #4181.  Can you try it out?

**Comment by leeping** [comment]
Thanks so much. I will try it now.

**Comment by leeping** [comment]
I changed my run parameters following @sef43 's suggestion to using a nonbonded cutoff of 1.0.  The code from #4181 ran successfully without errors on 100 repeats, whereas the previous code failed on every single repeat.  I call that a success. :)

I'm now running much longer trajectories and hopefully this time everything will be good.  Thanks so much for both of your attention to this issue!

**Comment by peastman** [comment]
Thanks!  I merged it.
